### ISSUE TITLE
Allow overriding the date

### ISSUE THREAD

**Issue by djmaze** [danielquinn/paperless#94]
Oftentimes a document's date is not correctly recognized during the OCR process, so it is automatically set to the current date. Especially for receipts, it would be good to additionally be able to manually set the date.

As the date when a document has been added can be useful as well, I propose adding an additional date field for this purpose.

**Comment by danielquinn** [comment]
Unfortunately, there's no way to detect document creation date.  At best, we could attempt to match a series of date regexes against the text found in a document, but there's no way to know which date (if any) refer to the actual date you want.  A phone bill for example could contain dozens, even hundreds of dates, and Paperless can't know which to use.  The consumer uses the file creation time when setting the `created` property rather than the time at which it was consumed, but for most cases, these times are practically identical.

When I first imported a bunch of old documents, I actually went through and used `touch` on each one to set the appropriate time so that Paperless would set the right one.  This worked well for all of my old mobile phone bills for example.  We've been talking about setting a creation time right in the name of the file to be parsed out before import, but I've been resisting that just because the name would be crazy-long and changing file modified time has the same effect.

So, TL;DR unless you can figure out how to make Paperless smart enough to discern the *right* date, I don't think this is something I can fix.

**Comment by djmaze** [comment]
First of all, thank you for the in-depth response.

> We've been talking about setting a creation time right in the name of the file to be parsed out before import, but I've been resisting that just because the name would be crazy-long and changing file modified time has the same effect.

Because I don't like to buy a document scanner just for personal use, I experimented with using a scanner app on my Android smartphone. The results are quite acceptable. The plan would be to allow uploading files to the consume folder of paperless (e.g. via WebDAV). 

On Android, it is not feasible to change file creation times (which would be needed when scanning older documents). Also, changing file creation time is quite a low-level OS operation which we probably shouldn't force users to do. So I think adding the date to the filename so it can be parsed is a valid approach in many cases. 

Also, you want to able to assign the _sending date_ instead of the arrival date, especially for receipts where the exact sending date plays an important role.

So, why not just let the user adjust the date in the UI? In my opinion it should be separate to the existing `created` field. Maybe just call it `date`?.

**Comment by danielquinn** [comment]
That is an excellent point.  I've just pushed an update that allows modification of the created time (I'm not sure why I disabled it in the first place) so if you do a `git pull` you should have that working.  I'm also looking into allowing it into the file name -- though this may be more difficult than it sounds, so I might not do it.

**Comment by timwhite** [comment]
I have existing documents that have the date in the filename, so some way to pull the date out of filenames would be very useful

**Comment by danielquinn** [comment]
I've just merged in a long-time PR (#79) that adds support for dates in the file name.  It's all documented [here](http://paperless.readthedocs.org/en/latest/consumption.html#a-note-on-file-naming).

I don't think I'll be hacking in support for guessing the date from the file name, since there's just too many permutations in there and no way to differentiate a date from a title or correspondent, or both.

**Comment by djmaze** [comment]
@danielquinn Thanks for the changes!

[According to the specs](https://github.com/danielquinn/paperless/blob/master/src/documents/tests/test_consumer.py#L138-L141), the more convenient date-only format `YYYYMMDDZ` is also supported. Should probably be documented as well.

**Comment by danielquinn** [comment]
Good point, and thanks.  All fixed :-)
